include/core/utilities.h

```diff
@@ -615,4 +617,26 @@ compute_cell_diameter(const double cell_measure, const unsigned int fe_degree)
   return h;
 }
 
+/**
+ * @brief Computes the volume of the cell by integrating after applying
+ * quadrature using a summation of JxW values over a cell.
+ *
+ * @tparam dim Number of spatial dimensions (2D or 3D).
+ *
+ * @param[in] fe_values FEValues object of the physics.
+ *
+ * @param[in] n_q_points Number of quadrature points.
+ *
+ * @return Volume (3D) or area (2D) of the cell.
+ */
+template <int dim>
+inline double
+compute_volume_with_JxW(const FEValues<dim> &fe_values, unsigned int n_q_points)
+{
+  double cell_volume = 0;
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    cell_volume += fe_values.JxW(q);
```

Comment:
Maybe using get_JxW_values() outside of the loop could be faster than calling JxW multiple times, but I'm not sure about the performance gains here

Reply:
Why not instead of providing the FeValues you could provide instead the vector of JxW. I don,t like the fact that you need to provide the loop limit (n_q_points) AND the fe_values. This is a bit weird.
You could just provide the vector of JxW and then just pass the vector to the function. The size of the vector would also give you the number of quadrature points.

Reply:
You're both right! I kind of just forgot about the vector. I'll change that! :)